:zap: workflowPreview component

Bearbobs opened the pull request.

janober committed:
:arrow_up: Update package-lock.json file

janober committed:
:whale: Extend nightly build action to also allow to build from other
repository

janober committed:
:whale: Extend to allow automatic master merge

janober committed:
:whale: Fix automerge

janober committed:
:whale: Another automerge fix

janober committed:
:whale: Try again to fix merge

Bearbobs committed:
:zap: init workflowPreview compoennt

Bearbobs committed:
:sparkles: Add /workflows/demo route (#2745)

* :zap: added /workflows/demo route

* :zap: suggested improvements

* :zap: added n8n version in postmessage

* :zap: updated zoom menu styling

* :zap: updated component padding for zoomToFit

* :zap: suggested improvements

* :zap: moved getFixedNodeList to canvasHelpers

* :pencil: removed unused import and rebased from master

pemontto committed:
:bug: Fix confusing placeholder text (#2557)

Bearbobs committed:
:pencil: updated type for message event

Bearbobs committed:
:pencil: fixed formatting

mutdmour reviewed (commented):
go through outdated comments too please

mutdmour commented on packages/editor-ui/src/components/WorkflowPreview.vue:
```diff
@@ -0,0 +1,86 @@
+<script lang="ts">
+import Vue from 'vue';
+
+export default Vue.extend({
```
also add component name
```suggestion
export default Vue.extend({
   name: 'WorkflowPreview',
```

janober committed:
:sparkles: Add optional health check endpoint to worker (#2786)

* :sparkles: Add optional health check endpoint to worker

* :shirt: Fix lint issue

* :zap: Improve error output

harshil1712 committed:
:zap: Update Codex files (#2797)

Bearbobs committed:
:zap: suggested improvements

RicardoE105 committed:
:sparkles: Add Linear Trigger node (#2767)

* :sparkles: Linear Trigger

* :art: Replace PNG with SVG icon

Co-authored-by: Iván Ovejero <ivov.src@gmail.com>

michael-radency committed:
:sparkles: Add HaloPSA node (#2620)

* added node ui

* wip problems with auth

* updated authentication

* fixed linter error

* added haloPSA request function

* removed any return type

* fixed linter errors

* added CRUD functionalities

* updating branch from master

* updated create case for clients resource, added limit to getAll operation

* added required fields when creating clients and sites, added methods for fetching data to dynamicly populate  options when creating site or client

* added required fields for users and invoices when operation is create

* :hammer: Removed some commented code

* :bug: Fix bug in url formating

* :hammer: fixed plural resources, fixed main for loop

* :hammer: fix trailing coma

* :hammer: fix for wrong resource endpoints

* :hammer: fixed linter complain in Jenkings node

* :hammer: replace custom fields with predefined

* :hammer: updating resources optional fields

* :zap: Small improvement

* :hammer: replaced fixedCollection to collection in resources description

* :hammer: updated site and ticket descriptions, code clean up

* :hammer: fixed accordingly to PR review

* :zap: Improvements

* :zap: Improvements

* :zap: Fix capitalization

* :shirt: Fix trailing comma

* :construction: node changes accordingly to review

* :zap: lint errors fix

* :zap: Activate simplify option by default

* :zap: Fix some more issues

Co-authored-by: ricardo <ricardoespinoza105@gmail.com>
Co-authored-by: Iván Ovejero <ivov.src@gmail.com>
Co-authored-by: Jan Oberhauser <jan.oberhauser@gmail.com>

SchnapsterDog committed:
N8N-2923 Added SaveWorkflow Modal

SchnapsterDog committed:
N8N-2923 Fixed Linter Issue

RicardoE105 committed:
:zap: Add StatusCallback parameter - Twilio (#2801)

* :zap: Add StatusCallback parameter

* :zap: Fix type

Co-authored-by: Jan Oberhauser <jan.oberhauser@gmail.com>

ivov committed:
:sparkles: Add Zammad node (#2621)

* add zammad

* :zap: First pass

* :shirt: Fix lint

* :zap: Refactor user resource

* :zap: Refactor group resource

* :zap: Refactor ticket resource

* :zap: Minor improvements

* :zap: Set workaround for broken endpoints

* :shirt: Fix lint

* :zap: Fix credentials test

* :package: Update package-lock.json

* :zap: Change defaults for active

* :zap: Refactor creds

* :shirt: Fix lint

* :package: Update package-lock.json

* :zap: Make first and last name required

* :zap: Replace email with login

* :zap: Switch defaults to true

* :zap: Add custom fields to groups

* :zap: Add inactive entities to loaders

* :zap: Move email to optional fields

* :zap: Validate for empty article

* :fire: Remove `ticket:update` per feedback

* :package: Update package-lock.json

* :truck: Rename import

* :shirt: Fix lint

* :zap: Small improvements

* :zap: Improvements

Co-authored-by: quansenB <inaki.breinbauer@gmail.com>
Co-authored-by: ricardo <ricardoespinoza105@gmail.com>

janober committed:
:bug: Fix monaco editor resize as suggested in #2800

janober committed:
:zap: Sort nodes in expression editor relative to current node position

janober committed:
:bug: Deactivate unwanted list autofill #2798

tennox committed:
:zap: Fix that fixedCollection leaves empty arrays (#2792)

When deleting the single option inside of an option type,
an empty array was left behind and will be:
- shown as empty heading in UI
- leave an empty array in JSON

ivov committed:
:zap: Sanitize description in uProc node (#2776)

* :fire: Sanitize link

* :fire: Shorten description

janober committed:
:whale: Fix parameter name

janober committed:
:zap: Make it possible to get file from specific branch/tag/commit
via GitHub

janober committed:
:shirt: Fix lint issue

janober committed:
:arrow_up: Update package-lock.json file

janober committed:
:bookmark: Release n8n-nodes-base@0.161.0

janober committed:
:bookmark: Release n8n-editor-ui@0.130.0

janober committed:
:arrow_up: Set n8n-editor-ui@0.130.0 and n8n-nodes-base@0.161.0 on n8n

janober committed:
:bookmark: Release n8n@0.163.0

janober committed:
:bug: Fix getParentTypes bug which caused problems with OAuth

janober committed:
:bookmark: Release n8n@0.163.1

SchnapsterDog committed:
N8N-2858 Templates Telemetry Events

SchnapsterDog committed:
N8N-2853 Merging with N8N-2923 (Save Workflow Modal)

Bearbobs committed:
:pencil: renamed nodeView to nodeViewDetailsOpened

Bearbobs commented on packages/editor-ui/src/components/WorkflowPreview.vue:
```diff
@@ -0,0 +1,92 @@
+<template>
+	<iframe
+		:class="{ workflow: !this.nodeView, openNDV: this.nodeView }"
```
missed to push the fix, we renamed nodeView to nodeViewDetailsOpened

Bearbobs commented on packages/editor-ui/src/components/WorkflowPreview.vue:
```diff
@@ -0,0 +1,92 @@
+
+export default mixins(showMessage).extend({
+	name: 'WorkflowPreview',
+	props: ['workflow'],
```
JSON.stringify(workflow_object) so that we don't have to change much when we integrate web component

Bearbobs committed:
:pencil: get workflow prop as JSON

mutdmour commented on packages/editor-ui/src/plugins/i18n/locales/en.json:
```diff
@@ -1079,5 +1079,13 @@
+	"workflowPreview": {
+		"showError": {
+			"previewError": {
+				"message": "Unable to preview workflow",
+				"title": "Preview error"
```
no need for message
```suggestion
				"title": "Unable to preview workflow"
```

SchnapsterDog committed:
N8N-2853 Updated GetTemplateById query

SchnapsterDog committed:
Merge branch 'n8n-2938-templates-fe-implement-vue-demo' of https://github.com/n8n-io/n8n into n8n-2938-templates-fe-implement-vue-demo

SchnapsterDog committed:
Merge branch 'master' of https://github.com/n8n-io/n8n into n8n-2938-templates-fe-implement-vue-demo

SchnapsterDog committed:
Test

mutdmour committed:
fix issues
